src/rgw/driver/d4n/rgw_sal_d4n.cc

```diff
@@ -566,7 +565,17 @@ int D4NFilterObject::set_head_obj_dir_entry(const DoutPrefixProvider* dpp, optio
 	ldpp_dout(dpp, 10) << "D4NFilterObject::" << __func__ << "(): BlockDirectory set method failed for head object with ret: " << ret << dendl;
       }
     } else if (ret == 0) { // head object exists; update instead of overwrite
-      block.prevVersion = {block.version, block.deleteMarker};
+      if (!this->get_bucket()->versioning_enabled() && block.version == "null") { // null delete markers get overwritten but only if versioning is suspended
```

Comment:
The comment says that this code is for null delete markers, but there is no check for whether this is a delete marker or not. If this is for dirty blocks versioned case, then we do not need this at this point. Please clarify so that I can proceed accordingly